src/js/plugins/plugin.text_selection.js

```diff
@@ -0,0 +1,204 @@
+      if (window.getSelection) window.getSelection().removeAllRanges();
+    })
+  }
+
+  /**
+   * Stops page flipping on short click, allows text selction on longer mousedown
+   * @param {JQuery} $container
+   */
+  stopPageFlip($container) {
+    const $svg = $container.find('svg');
+    let longPressTimer;
+    this.interceptCopy($container);
+
+    $svg.on("mousedown", (event) => {
+      if ($(event.target).is(this.svgWordElement)) {
+        event.stopPropagation();
+        longPressTimer = window.setTimeout(() => {
+          $svg.one("mouseup", (event) => {
+            event.stopPropagation();
+            clearTimeout(longPressTimer);
+            this.deselectOnClick($container);
+          });
+        }, 250);
+      }
+    });
+  }
+
+  /**
+   * @param {number} pageIndex
+   * @param {JQuery} $container
+   */
+  async createTextLayer(pageIndex, $container) {
+    const $svgLayers = $container.find('.textSelectionSVG');
+    if ($svgLayers.length) return;
+    const XMLpage = await this.getPageText(pageIndex);
+    const XMLwidth = $(XMLpage).attr("width");
+    const XMLheight = $(XMLpage).attr("height");
+
+    const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+    svg.setAttribute("xmlns", "http://www.w3.org/2000/svg");
+    svg.setAttribute("viewBox", `0 0 ${XMLwidth} ${XMLheight}`);
+    $container.append(svg);
+    svg.setAttribute('class', 'textSelectionSVG');
+    svg.setAttribute('preserveAspectRatio', 'none');
+    $(svg).css({
+      "width": "100%",
+      "position": "absolute",
+      "height": "100%",
+      "top": "0",
+      "left": "0",
+    });
+
+    $(XMLpage).find("PARAGRAPH").each((i, paragraph) => {
+      // adding text element for each paragraph in the page
+      const paragSvg = document.createElementNS("http://www.w3.org/2000/svg", this.svgParagraphElement);
+      paragSvg.setAttribute("class", "BRparagElement");
+      const words = $(paragraph).find("WORD");
+      let wordHeightMax = 0;
+
+      for(let i = 0; i < words.length; i++) {
+        // adding tspan for each word in paragraph
+        const currWord = words[i];
+        // eslint-disable-next-line no-unused-vars
+        const [left, bottom, right, top] = $(currWord).attr("coords").split(',').map(parseFloat);
+        const wordHeight = bottom - top;
+        if(wordHeight > wordHeightMax) wordHeightMax = wordHeight;
+
+        const wordTspan = document.createElementNS("http://www.w3.org/2000/svg", this.svgWordElement);
+        wordTspan.setAttribute("class", "BRwordElement");
+        wordTspan.setAttribute("x", left.toString());
+        wordTspan.setAttribute("y", bottom.toString());
+        wordTspan.setAttribute("textLength", (right - left).toString());
+        wordTspan.setAttribute("lengthAdjust", "spacingAndGlyphs");
+        wordTspan.textContent = currWord.textContent;
+        paragSvg.appendChild(wordTspan);
+
+        // adding spaces after words except at the end of the paragraph
```

Comment:
We can fix this in a future PR

```suggestion
        // adding spaces after words except at the end of the paragraph
        // TODO: assumes left-to-right text
```

Reply:
done